src/snowflake/snowpark/_internal/analyzer/analyzer.py

```diff
@@ -592,6 +613,11 @@ def analyze(
             str(expr)
         )  # pragma: no cover
 
+    def internal_alias_extractor(self, expr: Expression) -> Expression:
+        if isinstance(expr, _InternalAlias):
+            return expr.child
+        return expr
+
```

Comment:
do you think we can make it part of the `analyze` function like the following way:
```python
if isinstance(expr, _InternalAlias):
    return self.analyze(expr.child)
```

Reply:
I do want to simplify it but if we follow what you suggest, it would ignore all expressions. We still need to allow the alias to get passed through `unary_expression_extractor` so that when the function is top level, we still add the alias.